haystack/components/converters/azure.py

```diff
@@ -95,14 +129,8 @@ def run(self, sources: List[Union[str, Path, ByteStream]], meta: Optional[List[D
             result = poller.result()
             azure_output.append(result.to_dict())
 
-            file_suffix = None
-            if "file_path" in bytestream.meta:
-                file_suffix = Path(bytestream.meta["file_path"]).suffix
-
-            document = AzureOCRDocumentConverter._convert_azure_result_to_document(result, file_suffix)
-            merged_metadata = {**bytestream.meta, **metadata}
-            document.meta = merged_metadata
-            documents.append(document)
+            docs = self._convert_tables_and_text(result=result, meta=metadata)
+            documents.extend(docs)
 
         return {"documents": documents, "raw_azure_response": azure_output}
```

Comment:
@vblagoje is this the first node that returns two types of documents? If so would it better to return them in two separate streams and not just under `documents`? I say this since we often want to treat tables differently than text documents (for example use a different embedder). What do you think?

Reply:
Sure, but we'll be breaking a signature of this component. That's the downside. I don't think many ppl are using it but still. We could return documents for all and then a few other outputs for specific types but that's somehow smelly to me. Wdyt?

Reply:
Ok, let's ping most and see who responds! Need some input here @masci @anakin87 @silvanocerza @julian-risch 🙏

Reply:
Discussing with @vblagoje he made the suggestion to keep documents output and create a custom component that could come after to be able to sort document types. I think this sounds like a good idea.